docs/administration/authentication-providers/okta-authentication.md

```diff
@@ -0,0 +1,156 @@
+:::
+
+:::success
+**Already have Octopus user accounts?**
+If you already have Octopus user accounts and you want to enable external authentication, simply make sure the Email Address matches in both Octopus and the external identity provider. This means your existing users will be able to sign in using an external identity provider and still belong to the same teams in Octopus.
+:::
+
+### Getting permissions
+
+!include <admin-user>
+
+## Troubleshooting {#Oktaauthentication-Troubleshooting}
+
+We do our best to log warnings to your Octopus Server log whenever possible. If you are having difficulty configuring Octopus to authenticate with Okta, be sure to check your [server logs](/docs/reference/log-files.md) for warnings.
+
+You can also check Okta logs by clicking the **View Logs** link on the Okta admin portal.
+
+![](/docs/images/okta-authentication/okta-view-logs.png "width=500")
+
+### Double and Triple check your configuration {#Oktaauthentication-DoubleandTriplecheckyourconfiguration}
+
+Unfortunately security-related configuration is sensitive to everything. Make sure:
+
+- you don't have any typos or copy-paste errors
+- remember things are case-sensitive
+- remember to remove or add slash characters as we've instructed - they matter too!
+
+### Check OpenID Connect metadata is working {#Oktaauthentication-CheckOpenIDConnectmetadataisworking}
+
+You can see the OpenID Connect metadata by going to the Issuer address in your browser adding `/.well-known/openid-configuration` to the end. In our example this would have been something like `https://dev-xxxxxx.oktapreview.com/.well-known/openid-configuration`
+
+### Inspect the contents of the security token {#Oktaauthentication-Inspectthecontentsofthesecuritytoken}
+
+Perhaps the contents of the security token sent back by Okta aren't exactly the way Octopus expected, especially certain claims which may be missing or named differently. This will usually result in the Okta user incorrectly mapping to a different Octopus User than expected. The best way to diagnose this is to inspect the JSON Web Token (JWT) which is sent from Okta to Octopus via your browser. To inspect the contents of your security token:
+
+1. Open the Developer Tools of your browser and enable Network logging making sure the network logging is preserved across requests.
+2. In Chrome Dev Tools this is called "Preserve Log".
+
+   ![](/docs/images/5670656/5866122.png)
+
+3. Attempt to sign into Octopus using Okta and find the HTTP POST coming back to your Octopus instance from Okta on a route like `/api/users/authenticatedToken/Okta`. You should see an `id_token` field in the HTTP POST body.
+4. Grab the contents of the `id_token` field and paste that into [https://jwt.io/](https://jwt.io/) which will decode the token for you.
+
+   ![](/docs/images/5670656/5866123.png "width=500")
+
+5. Don't worry if jwt.io complains about the token signature, it doesn't support RS256 which is used by Okta.
```

Comment:
Having this as its own dot point is a bit odd - it feels like it should be a call out or something, rather than a step in the process.